ibis/bigquery/client.py

```diff
@@ -305,24 +348,30 @@ def _build_ast(self, expr, context):
         return result
 
     def _execute_query(self, dml, async=False):
+        if async:
+            raise NotImplementedError(
+                'Asynchronous queries not implemented in the BigQuery backend'
```

Comment:
The `QueryJob` class does implement the `concurrent.futures` interface, but yeah no `async` keyword since `google-cloud-bigquery` has to support Python 2.7 still.

Reply:
Yeah, this was really to solidify the expectations about the `async` parameter here so that a reasonable error message would be visible to users if they pass `async=True`.